src/core/reference/include/openvino/reference/fake_quantize.hpp

```diff
@@ -55,135 +110,396 @@ void fake_quantize(const T* const arg,
+                                         const Shape& arg_shape,
+                                         T levels_minus_one,
+                                         size_t input_inner_stride,
+                                         const F& get_outer_strides) {
+    fake_quantize_loop(arg_shape,
+                       arg,
+                       in_low,
+                       in_high,
+                       out_low,
+                       out_high,
+                       out,
+                       input_inner_stride,
+                       get_outer_strides,
+                       [levels_minus_one](const T* input,
+                                          const T* const input_end,
+                                          const T* in_low,
+                                          const T* in_high,
+                                          const T* out_low,
+                                          const T* out_high,
+                                          T* out) {
+                           transform(input,
+                                     input_end,
+                                     in_low,
+                                     in_high,
+                                     out_low,
+                                     out_high,
+                                     out,
+                                     [levels_minus_one](T input, T in_low, T in_high, T out_low, T out_high) {
+                                         return quantize(input, in_low, in_high, out_low, out_high, levels_minus_one);
+                                     });
+                       });
+}
+
+template <typename T, typename F>
+void fake_quantize_unit_inner_stride(const T* arg,
+                                     const T* in_low,
+                                     const T* in_high,
+                                     const T* out_low,
+                                     const T* out_high,
+                                     T* out,
+                                     const Shape& arg_shape,
+                                     T levels_minus_one,
+                                     size_t input_inner_stride,
+                                     const F& get_outer_strides) {
+    auto quantize_with_scalar_intervals = [levels_minus_one](const T* input,
+                                                             const T* const input_end,
+                                                             const T* in_low,
+                                                             const T* in_high,
+                                                             const T* out_low,
+                                                             const T* out_high,
+                                                             T* out) {
+        auto in_low_scalar = *in_low;
+        auto in_high_scalar = *in_high;
+        auto out_low_scalar = *out_low;
+        auto out_high_scalar = *out_high;
```

Comment:
Can those variables be marked as const? General comment to add missing `const` where possible to improve readability.
```suggestion
        const auto in_low_scalar = *in_low;
        const auto in_high_scalar = *in_high;
        const auto out_low_scalar = *out_low;
        const auto out_high_scalar = *out_high;
```

Reply:
done